### ISSUE TITLE
Wrong query when getting user on facebook login

### ISSUE THREAD

**Issue by pragone** [strapi/strapi#2536]
**Informations**
- **Node.js version**: 
- **NPM version**: v10.14.2
- **Strapi version**: 3.0.0-alpha.17
- **Database**: MySQL
- **Operating system**: Mac

**What is the current behavior?**
I'm trying the login-react example and configured login with Facebook. (BTW, now Facebook requires the url for the redirect to be SSL enabled, so I just used https://www.npmjs.com/package/local-ssl-proxy to create a local SSL proxy with `local-ssl-proxy --source 4337 --target 1337` and configured the Proxy in Configurations -> Server -> Proxy Enable: On, Proxy Host: localhost, Proxy Port 4337, Proxy SSL: On. Hope that helps others)

The OAuth flow is mostly working, but in the final call of the flow where the access_token is provided back from Facebook and the backend queries to get the user based on the email I'm getting a 400 Bad Request.

**Steps to reproduce the problem**

* Get a strapi server up and running and configure the Facebook Provider in it.
* Try logging in with the login-react example. You'll see the last request to be something like: `https://localhost:4337/auth/facebook/callback?access_token=SSHHHSECRETBAOdDq4lZC6uXzvT75RyjI9zCUNEHRQeH9EBngt3dc5kZBt1ZBQuYNcHAYQgz9RlM2gMw0lWARf04PwfZBNIM4Iz1q0lXRBhHGZCCZCoDHGV9MpLYcBBLRbLZBxzQXBuGxXMRvGlEbUJg2OJPUXYgCsjDSEnZCptOZCAZDZD&raw%5Baccess_token%5D=SECRETTOOAOdDq4lZC6uXzvT75RyjI9zCUDDDDDDDDDDDBQuYNcHAYQgz9RlM2gMw0lWARf04PwfZBNIM4Iz1q0lXRBhHGZCCZCoDHGV9MpLYcBBLRbLZBxzQXBuGxXMRvGlEbUJg2OJPUXYgCsjDSEnZCptOZCAZDZD&raw%5Btoken_type%5D=bearer&raw%5Bexpires_in%5D=5183445#_=_`
* The request returns a 400 Bad Request

**What is the expected behavior?**
It should log you in

**Suggested solutions**
After some debugging (which, BTW was difficult because the exception was swallowed and not logged anywhere), I realised that the issue was that when the user-permissions module is trying to load the user it's having issues with building the query.
This was the trace that gave me the clue (which I got by adding logging in plugins/users-permissions/services/Providers.js in line 91, in the catch):
```
TypeError: The operator "undefined" is not permitted
    at Formatter.operator (/Users/pragone/code/plan4event-api/node_modules/knex/lib/formatter.js:179:13)
    at QueryCompiler_MySQL.whereBasic (/Users/pragone/code/plan4event-api/node_modules/knex/lib/query/compiler.js:567:100)
    at QueryCompiler_MySQL.where (/Users/pragone/code/plan4event-api/node_modules/knex/lib/query/compiler.js:350:34)
    at components.map.component (/Users/pragone/code/plan4event-api/node_modules/knex/lib/query/compiler.js:96:67)
    at Array.map (<anonymous>)
    at QueryCompiler_MySQL.select (/Users/pragone/code/plan4event-api/node_modules/knex/lib/query/compiler.js:96:35)
    at QueryCompiler_MySQL.toSQL (/Users/pragone/code/plan4event-api/node_modules/knex/lib/query/compiler.js:50:29)
    at Builder.toSQL (/Users/pragone/code/plan4event-api/node_modules/knex/lib/query/builder.js:72:44)
    at /Users/pragone/code/plan4event-api/node_modules/knex/lib/runner.js:37:34
    at tryCatcher (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/util.js:16:23)
    at /Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/using.js:185:26
    at tryCatcher (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:694:18)
    at Promise._fulfill (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:638:18)
    at PromiseArray._resolve (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise_array.js:126:19)
    at PromiseArray._promiseFulfilled (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise_array.js:144:14)
    at Promise._settlePromise (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:574:26)
    at Promise._settlePromise0 (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/promise.js:694:18)
    at _drainQueueStep (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/pragone/code/plan4event-api/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)``` 

I'll submit a PR for it.

**Comment by pragone** [comment]
Just to add a bit here. I think I have a better understanding of the issue.
I think that the main issue is that the mongoose backend supports passing a filter like:

```
const users = await strapi.query('user', 'users-permissions').find({
           where: {	           
             email: profile.email
           }
         });
```

while the same query fails for bookshelf because it doesn't specify the `symbol` (meaning the '=') This is not specific to the social login feature. It just happens that it does a query in that way.

I think that the best solution would be to make bookshelf understand this type of queries. I added a comment on the PR with what the code would be. If you agree that that is the right approach I'll create a pull request for that change and discard this one.

**Comment by pragone** [comment]
Actually... just checked. Most times it'll be fine because the filters passed to 'find' are the result of a `convertParams` method call... but in this case we're just building the query manually... So I think that the fix as submitted in the associated PR is probably the right one. Meaning, let's just change how the user-permissions plugin constructs that query.

**Comment by pragone** [comment]
@lauriejim Could you have a read at these comments. I think that it'll be easier to reproduce after you read the actual cause, and therefore the PR would be clearer

**Comment by AndinAvdiu** [comment]
Is this solved? because i'm having this problem also. Should it be "the right way" if I modify the users-permission plugin?

**Comment by pragone** [comment]
Hey @AndinAvdiu . The fix has been merged, but I think it'll be released in 3.0.0-alpha.19
Modifying the plugin is not the right way in that it'll be overwritten every time the post-install scripts run. But it'll solve your issue in the meantime, and 3.0.0-alpha.19 shouldn't be far away.
Good luck

**Comment by AndinAvdiu** [comment]
Make sure to make getProfile method in the "Providers" service accessible.
